File: market/src/main/java/com/challenge/market/member/service/MemberService.java
package com.challenge.market.member.service;

import com.challenge.market.member.domain.Member;
import com.challenge.market.member.repository.MemberRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class MemberService {

    private final MemberRepository memberRepository;


    /**
     * 회원 가입
     *
     * @param member
     * @return
     */
    @Transactional(readOnly = false)
    public Member signUp(Member member) {
        return memberRepository.save(member);
    }
}

File: api/src/main/kotlin/wanted/market/api/domain/member/service/MemberService.kt
package wanted.market.api.domain.member.service

import jakarta.transaction.Transactional
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import wanted.market.api.domain.member.dto.RegisterMemberRequest
import wanted.market.api.domain.member.dto.LoginMemberRequest
import wanted.market.api.domain.member.entity.Member
import wanted.market.api.domain.member.repository.MemberRepository

@Service
@Transactional
class MemberService(
    @Autowired private val memberRepository: MemberRepository
) {
    fun registerMember(registerMemberRequest: RegisterMemberRequest) {
        val newMember = Member(
            email = registerMemberRequest.email,
            password = registerMemberRequest.password
        )
        memberRepository.save(newMember)
    }

    fun loginMember(loginMemberRequest: LoginMemberRequest) {
        memberRepository.findByEmailAndPassword(email= loginMemberRequest.email, password= loginMemberRequest.password)
    }

    fun findMember(memberId: Long): Member {
        val member = memberRepository.findById(memberId)
            .orElseThrow{ RuntimeException() }
        return member
    }
}
